zstd/seqdec_amd64.go

```diff
@@ -101,3 +101,104 @@ func (s *sequenceDecs) decode(seqs []seqVals) error {
 	}
 	return err
 }
+
+type executeAsmContext struct {
+	seqs        []seqVals
+	seqIndex    int
+	out         []byte
+	literals    []byte
+	outPosition int
+	litPosition int
+}
+
+// sequenceDecs_executeSimple_amd64 implements the main loop of sequenceDecs.executeSimple in x86 asm.
+//
+// Please refer to seqdec_generic.go for the reference implementation.
+//go:noescape
+func sequenceDecs_executeSimple_amd64(ctx *executeAsmContext) bool
+
+// executeSimple handles cases when no history nor dictionary are used.
+func (s *sequenceDecs) executeSimple(seqs []seqVals) error {
+	// Ensure we have enough output size...
+	if len(s.out)+s.seqSize > cap(s.out) {
```

Comment:
ok, so either here, or where you decide to call, you need to take the over-copies into account, so we never write outside the cap of the slice we are given.

Reply:
Right, added extra padding and removed checks from the asm procedure.